packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBarRow.tsx

```diff
@@ -542,6 +546,40 @@ export class UnthemedSpanBarRow extends React.PureComponent<SpanBarRowProps> {
       </TimelineRow>
     );
   }
+
+  getSpanBarLabel = (span: TraceSpan, spanBarOptions: SpanBarOptions | undefined, duration: string) => {
+    const type = spanBarOptions?.type ?? '';
+
+    if (type === NONE) {
+      return '';
+    } else if (type === '' || type === DURATION) {
+      return `(${duration})`;
+    } else if (type === TAG) {
+      const tagKey = spanBarOptions?.tag?.trim() ?? '';
+      if (tagKey !== '' && span.tags) {
+        const tag = span.tags.filter((tag: TraceKeyValuePair) => {
```

Comment:
I think `find` would be a more appropriate function since we are trying to find the first instance of the tag. Same for in process below.